activerecord/lib/active_record/core.rb

```diff
@@ -680,21 +683,18 @@ def connection_handler
       self.class.connection_handler
     end
 
-    # Returns the contents of the record as a nicely formatted string.
+    # Returns the attributes specified by <tt>.attributes_for_inspect</tt> as a nicely formatted string.
     def inspect
-      # We check defined?(@attributes) not to issue warnings if the object is
-      # allocated but not initialized.
-      inspection = if defined?(@attributes) && @attributes
-        attribute_names.filter_map do |name|
-          if _has_attribute?(name)
-            "#{name}: #{attribute_for_inspect(name)}"
-          end
-        end.join(", ")
+      if attributes_for_inspect == :all
+        inspect_with_attributes(attribute_names)
```

Comment:
```suggestion
        full_inspect
```

Reply:
Agree

Reply:
Argh, thanks. I think I left it like this when I was playing with a version that had `inspect(:all)` instead of `full_inspect`